### PR TITLE
One page UI

### PR DESCRIPTION
**Pull request by LakshitXD**

### CODE REVIEW

File: devsnest/db/migrate/20210614160228_add.rb [hunk 1]
```diff
@@ -0,0 +1,5 @@
+class Add < ActiveRecord::Migration[6.0]
+  def change
+    add_column :users, :google_id, :string
```
Comment (vedansh) [inline devsnest/db/migrate/20210614160228_add.rb:3]: remove this file

File: devsnest/db/migrate/20210618132309_add_bot_token_to_users.rb [hunk 2]
```diff
@@ -1,4 +1,4 @@
-class AddGoogleIdToUsers < ActiveRecord::Migration[6.0]
+class AddBotTokenToUsers < ActiveRecord::Migration[6.0]
```
Comment (vedansh) [inline devsnest/db/migrate/20210618132309_add_bot_token_to_users.rb:1]: do not change this

File: devsnest/db/migrate/20210618132007_add_video_questions_to_contents.rb [hunk 3]
```diff
@@ -0,0 +1,5 @@
+class AddVideoQuestionsToContents < ActiveRecord::Migration[6.0]
+  def change
+    add_column :contents, :video_questions, :text
```
Comment (vedansh) [inline devsnest/db/migrate/20210618132007_add_video_questions_to_contents.rb:3]: json

File: devsnest/app/resources/api/v1/content_resource.rb [hunk 4]
```diff
@@ -15,6 +15,10 @@ def self.default_sort
         [{ field: 'priority', direction: :asc }]
       end
 
+      def questions_list
+        Content.where(id: video_questions.map(&:to_i)).as_json
```
Comment (vedansh) [inline devsnest/app/resources/api/v1/content_resource.rb:19]: also add status, from submission

Comment (vedansh) [reply devsnest/app/resources/api/v1/content_resource.rb:19]: res = []
Content.where(id: video_questions.map(&:to_i)).each do |c|
     sub = Submission.where(user_id: contenxt[:user].id, content_id: c.id).first
 
     res.push c.as_json.merge(status: sub.exists? ? c.status: 0
end
res

File: devsnest/app/resources/api/v1/content_resource.rb [hunk 5]
```diff
@@ -15,6 +15,18 @@ def self.default_sort
         [{ field: 'priority', direction: :asc }]
       end
 
+      def questions_list
+        contents = Content.where(id: video_questions)
+        return {} if video_questions.nil?
```
Comment (vedansh) [inline devsnest/app/resources/api/v1/content_resource.rb:20]: return []

File: devsnest/app/resources/api/v1/content_resource.rb [hunk 6]
```diff
@@ -15,6 +15,18 @@ def self.default_sort
...
+        questions = []
+        contents.each do |c|
+          sub = Submission.where(user_id: context[:user].id, content_id: c.id).first
+
+          questions.push c.as_json.merge(status: sub != nil ? sub.status: "notdone")
```
Comment (vedansh) [inline devsnest/app/resources/api/v1/content_resource.rb:25]: `questions.push c.as_json.merge(status: sub.nil? ? sub.status: "notdone")`

File: devsnest/db/migrate/20210614160228_add.rb (hunk 1 above)
Comment (vedansh) [reply devsnest/db/migrate/20210614160228_add.rb:3]: Ye remove kar do

File: devsnest/db/migrate/20210618132309_add_bot_token_to_users.rb (hunk 2 above)
Comment (vedansh) [reply devsnest/db/migrate/20210618132309_add_bot_token_to_users.rb:1]: ye bhi

File: devsnest/db/migrate/20210614160228_add.rb (hunk 1 above)
Comment (LakshitXD) [reply devsnest/db/migrate/20210614160228_add.rb]: done

File: devsnest/db/migrate/20210618132309_add_bot_token_to_users.rb (hunk 2 above)
Comment (LakshitXD) [reply devsnest/db/migrate/20210618132309_add_bot_token_to_users.rb]: done

File: devsnest/app/resources/api/v1/content_resource.rb (hunk 6 above)
Comment (LakshitXD) [reply devsnest/app/resources/api/v1/content_resource.rb]: done

**Review by kaydee0502 — APPROVED** [review]

**Pull request merged**